modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -191,6 +211,42 @@ namespace cv {
                     fused_layer_names.back() = last_layer;
                 }
 
+                void setSwish()
+                {
+                    cv::dnn::LayerParams activation_param;
+                    activation_param.name = "Swish-name";
+                    activation_param.type = "Swish";
+
+                    darknet::LayerParameter lp;
+                    std::string layer_name = cv::format("swish_%d", layer_id);
+                    lp.layer_name = layer_name;
+                    lp.layer_type = activation_param.type;
+                    lp.layerParams = activation_param;
+                    lp.bottom_indexes.push_back(last_layer);
+                    last_layer = layer_name;
+                    net->layers.push_back(lp);
+
+                    fused_layer_names.back() = last_layer;
+                }
+
+                void setLogistic()
+                {
+                    cv::dnn::LayerParams activation_param;
+                    activation_param.name = "Sigmoid-name";
```

Comment:
Please check if we really need to specify names here. Due they are not unique, maybe they are not used at all?

Reply:
They are not used.

The actual name and type are set in `darknet::LayerParameter`. This name and type is used in `darknet_importer.cpp` to add layers using `Net::addLayer`. `Net::addLayer` invokes `LayerData` constructor which accepts name, type and `Net::LayerParams`. In the constructor, it overwrites the `name` and `type` field of `Net::LayerParams` with the name and type that were passed as arguments.

Hence, these names `LayerType-name` which are used through the darknet importer are never used.

Reply:
@dkurt Should I remove them? If yes, for all the layers which aren't part of this PR or only the layers which are part of this PR?

Reply:
Maybe just for newly created layers, thanks!